app/reaction.go

```diff
@@ -51,6 +52,16 @@ func (a *App) SaveReactionForPost(reaction *model.Reaction) (*model.Reaction, *m
 	// The post is always modified since the UpdateAt always changes
 	a.invalidateCacheForChannelPosts(post.ChannelId)
 
+	if pluginsEnvironment := a.GetPluginsEnvironment(); pluginsEnvironment != nil {
+		a.Srv().Go(func() {
+			pluginContext := a.PluginContext()
+			pluginsEnvironment.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
+				hooks.ReactionHasBeenAdded(pluginContext, reaction)
```

Comment:
Traditionally, we pass a minimal amount of information down to the plugin -- in this case, just the reaction. However, I have a feeling most plugins will immediately turn around to fetch the post and channel, which we conveniently already have when both adding and deleting reactions.

Should we consider including them here preemptively? Or perhaps a subset of the most actionable information, such as `channel.TeamID`, `channel.Id`, `post.UserId`, and maybe `post.RootId`? This doesn't enable anything the plugin can't already do, but might make this path significantly more performant for the likely use case of turning around to query this information.

Reply:
0/5 we stick with the existing pattern (minimal).

Reply:
@lieut-data - great point!  For the plugin that I'm working on, I'm having to do exactly as you described (going out and fetching the post to find out who the post author is), but I can also see the performance computational cost of the serialization required by RPC (I haven't looked deep into the plugin framework, but my assumption is that there is some serialization taking place on either end). 

If there is indeed serialization occurring, then aside from the reaction itself, if we stick with basic types then that might be a good middle ground to (a) helping out plugins, while (b) minimizing the overhead. Or, I see some 👍 on the sticking with minimal... 

Any other thoughts on this, or should we leave it minimal?

Reply:
I would throw it out there: What is easiest for a new developer when building a plugin/app?  

Personally, in the scenarios with DeletedReaction possibly happening, I'd want to verify on the parent post which actual emojis are currently there when displaying a vote count for example.  In case an event was missed for some reason.  For that example, I think it's best if we return the additional contextual `post.rootid`, etc.    0/5

Reply:
/If/ every plugin will turn around and fetch the data themselves, then the serialization overhead occurs anyway, plus the cost of querying the database again, etc. Otherwise, you're right: the overhead would be a waste.

Let's go with @levb's proposal to stick with the existing pattern for now unless we measure performance issues otherwise.

Reply:
Thanks everyone! Leaving as-is :)